api/src/main/java/io/opentelemetry/trace/Link.java

```diff
@@ -23,10 +23,16 @@
 import javax.annotation.concurrent.Immutable;
 
 /**
- * A link to a {@link Span} from a different trace.
+ * A link to a {@link Span}.
  *
- * <p>Used (for example) in batching operations, where a single batch handler processes multiple
- * requests from different traces.
+ * <p>It can be used in the following scenarios:
+ *
+ * <ul>
+ *   <li>"Fork-join" scenario. Link together spans from the same trace after a concurrent
```

Comment:
I am not sure that this is how we should model a fork-join. I kind of think that the parent should wait for all the spawned children. So I suggest to not put this as an example because users will model these operations like that.

Reply:
I will think about other example.

Do you have something in mind?

Reply:
I have reverted it and removed "different"